### PR TITLE
Add code to list filtered GitHub tags

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/14238

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/internal/github/github.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package github
+
+import (
+	"context"
+
+	"github.com/giantswarm/config-controller/service/github/internal/graphql"
```
Comment (corest) [inline service/internal/github/github.go:6]:
```suggestion
	"github.com/giantswarm/config-controller/service/internal/github/internal/graphql"
```

Comment (corest) [reply service/internal/github/github.go:6]: Also, for my knowledge. Why are there so many nested `internal`  folders? And why don't we put github/graphql package into `pkg`? As it is going to be used in cli.

Comment (kopiczko) [reply service/internal/github/github.go]: Thanks. Maybe I went a little bit wild. I'll move to `/pkg` then.

**Review by corest — APPROVED** [review]
lgtm

**Pull request merged**